rdt/hyper_transformer.py

```diff
@@ -381,6 +381,7 @@ def _validate_all_fields_fitted(self):
 
     def _sort_output_columns(self):
         """Sort ``_output_columns`` to follow the same order as the ``_input_columns``."""
+        self._output_columns = []
```

Comment:
Since this method uses `get_final_output_columns` which uses the `_transformers_tree`, I think we need to reset the `_transformers_tree` at the beginning of `fit`. Otherwise, if we fit onto data with different output columns, I think the output columns from the previous fit will show up as well.